File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums){

        int max = nums[0];
        int n = nums.length;
        int sum = 0;

        for(int i=0; i<nums.length; i++){
            sum = sum + nums[i];
            max = max>sum?max:sum;
            if(sum<0)
                sum = 0;
        }    
    return max;
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:            
        max_value = nums[0]
        sum_value = 0

        for n in nums:
            if sum_value < 0:
                sum_value = 0

            sum_value += n
            max_value = max(max_value, sum_value)
        
        return max_value